File: app/src/main/java/com/andyanika/translator/di/ScreenRouterImpl.java
package com.andyanika.translator.di;

import com.andyanika.translator.common.interfaces.ScreenRouter;

import ru.terrakok.cicerone.Router;

public class ScreenRouterImpl implements ScreenRouter {
    private Router router;

    ScreenRouterImpl(Router router) {
        this.router = router;
    }

    @Override
    public void navigateTo(String screenKey, Object data) {
        router.navigateTo(screenKey, data);
    }

    @Override
    public void backTo(String screenKey) {
        router.backTo(screenKey);
    }
}

File: app/src/main/java/com/andyanika/translator/di/ScreenRouterImpl.kt
package com.andyanika.translator.di

import com.andyanika.translator.common.constants.Screens
import com.andyanika.translator.common.interfaces.ScreenRouter
import com.github.terrakok.cicerone.Router
import com.andyanika.translator.main.Screens as ScreenFragments

class ScreenRouterImpl internal constructor(private val router: Router) : ScreenRouter {
    override fun navigateTo(screenKey: String, args: Any) {
        router.navigateTo(getScreenByKey(screenKey, args))
    }

    override fun backTo(screenKey: String) {
        router.backTo(getScreenByKey(screenKey))
    }

    fun getScreenByKey(screenKey: String, args: Any? = null) = when (screenKey) {
        Screens.TRANSLATION -> ScreenFragments.Translation()
        Screens.HISTORY -> ScreenFragments.History()
        Screens.FAVORITES -> ScreenFragments.Favorite()
        Screens.SELECT_LANGUAGE -> ScreenFragments.SelectLanguage(args as String)
        else -> throw IllegalAccessException(screenKey)
    }
}
